packages/shared/src/components/plus/PlusIOS.tsx

```diff
@@ -0,0 +1,196 @@
+                product.attributes.offerName
+              ] as PlusPriceType;
+              const price = parseInt(product.attributes.offers[0].price, 10);
+              const monthlyPrice = +(
+                price / (duration === PlusPriceType.Yearly ? 12 : 1)
+              ).toFixed(2);
+              const currencySymbol =
+                product.attributes.offers[0].priceFormatted.replace(
+                  /\d|\.|\s|,/g,
+                  '',
+                );
+
+              return {
+                label: product.attributes.name,
+                value: product.attributes.offerName,
+                price: {
+                  amount: product.attributes.offers[0].price,
+                  formatted: product.attributes.offers[0].priceFormatted,
+                  monthlyAmount: monthlyPrice,
+                  monthlyFormatted: `${currencySymbol}${priceFormatter.format(
+                    monthlyPrice,
+                  )}`,
+                },
+                currencyCode: product.attributes.offers[0].currencyCode,
+                currencySymbol,
+                extraLabel: product.attributes?.description?.standard,
+                appsId: PlusPriceTypeAppsId.Default,
+                duration,
+              };
+            })
+            .sort((a: { value: string }, b: { value: string }) => {
+              // Make sure that the products are sorted in the same order as the product list
+              // because the native code does not guarantee the order of the products
+              const aIndex = productList.indexOf(a.value);
+              const bIndex = productList.indexOf(b.value);
+              return aIndex - bIndex;
+            });
+        },
+      );
+
+      globalThis.webkit.messageHandlers[
+        WebKitMessageHandlers.IAPProductList
+      ].postMessage(productList);
+
+      return products;
+    },
+  });
+
+  useEffect(() => {
+    promisifyEventListener('iap-purchase-result', (event) => {
+      const result = event.detail;
+      if (result !== 'success') {
+        return;
+      }
+
+      router.replace(`${webappUrl}plus/success`);
+    });
+  }, [router, selectedOption]);
+
+  return (
+    <div
+      className="flex flex-col p-6"
+      ref={(element) => {
+        if (!element) {
+          return;
+        }
+
+        if (productOptions?.[0]?.value && !selectedOption) {
+          setSelectedOption(productOptions?.[0]?.value);
+        }
+      }}
+    >
+      <div className="flex gap-2">
+        <Button
+          onClick={() => {
+            query.refetch();
+          }}
+          variant={ButtonVariant.Float}
+        >
+          Reload
+        </Button>
+        <Button
+          onClick={() => {
+            globalThis.webkit.messageHandlers[
+              WebKitMessageHandlers.IAPSubscriptionManage
+            ].postMessage(null);
+          }}
+          variant={ButtonVariant.Float}
+        >
+          Manage
+        </Button>
+      </div>
```

Comment:
Will remove these before merging to main. Right now the PR targets an intermediate branch.